prusti-interface/src/parser.rs

```diff
@@ -2162,15 +2257,34 @@ impl<'tcx> Folder for SpecParser<'tcx> {
                                 item.span,
                                 PRUSTI_SPEC_ATTR,
                                 &specid.to_string(),
-                ));
+                        ));
                     }
 
-                    let mut new_trait_items = vec![];
+                    let trait_symbols: HashSet<Symbol> = trait_items
+                        .clone()
+                        .into_iter()
+                        .map(|item| item.ident.name)
+                        .collect();
+
+                    for symbol in funcs_symbols {
+                        if !trait_symbols.contains(&symbol) {
+                            self.report_error(
+                                item.span,
+                                &format!("no method '{}' in super trait", symbol));
+                        }
+                    }
 
+                    let mut new_trait_items = vec![];
                     for trait_item in trait_items.into_iter() {
                         match trait_item.node {
                             ast::TraitItemKind::Method(..) => {
-                                new_trait_items.extend(self.rewrite_trait_item_method(trait_item));
+                                let method_symbol = trait_item.ident.name.clone();
+                                let func_ref = (trait_symbol.clone(), method_symbol);
+                                let reg = self.register.lock().unwrap();
+                                let func_attrs = reg.get_attrs_refine(&func_ref);
+                                drop(reg);
```

Comment:
I would prefer to add a scope instead of having to call `drop`.

e.g. would this work?
```
let func_attrs = {
    let reg = self.register.lock().unwrap();
    reg.get_attrs_refine(&func_ref)
};
```